spec/models/term_spec.rb

```diff
@@ -0,0 +1,156 @@
+require 'rails_helper'
+
+RSpec.describe Term, type: :model do
+  let(:term_solr_doc) { URIService.solr.find_term(term.vocabulary.string_key, term.uri) }
+
+  describe 'when creating a external term' do
+    let(:term) { FactoryBot.create(:external_term) }
+
+    it 'adds uri_hash' do
+      expect(term.uri_hash).to eql '37e37aa82a659464081b368efa3f06dc12e56bd07ed8237f4c4a05f401015e52'
+    end
+
+    it 'add uuid' do
+      expect(term.uuid).not_to be blank?
+    end
+
+    it 'return error when uri invalid' do
+      expect {
+        FactoryBot.create(:external_term, uri: 'ldfkja')
+      }.to raise_error ActiveRecord::RecordInvalid
+    end
+
+    it 'returns error if uri is missing' do
+      expect { FactoryBot.create(:external_term, uri: '') }.to raise_error ActiveRecord::RecordInvalid
+    end
+
+    it 'returns error if uri is already represented in vocabulary' do
+      term
+      expect { FactoryBot.create(:external_term) }.to raise_error ActiveRecord::RecordInvalid
+    end
+
+    it 'creates solr document' do
+      expect(term_solr_doc).to include(
+        'uri' => 'http://id.worldcat.org/fast/1161301/',
+        'pref_label' => 'Unicorns',
+        'term_type' => 'external',
+        'authority' => 'fast',
+        'custom_fields' => '{"harry_potter_reference":true}',
+      )
+    end
+  end
+
+  describe 'when creating a local term' do
+    let(:term) { FactoryBot.create(:local_term) }
+
+    it 'add local term uri' do
+      expect(term.uri).to start_with 'https://example.com/term/'
```

Comment:
Do you mind adding an additional check here that the term starts with `Rails.application.config.local_uri_host`? That would be informative for anyone looking at the tests who doesn't know the backing logic as well.

Reply:
Sure, tho it would be a different look up if we store that in the uri_service.yml.